### DISCUSSION TITLE
Does minIO server support deep copy?

### DISCUSSION THREAD

**Discussion by Wzing0421** [minio/minio#11460] [Q&A]
I have a question about minIO :
If it supports amazon s3 api, can it support deep copy? I mean, in the worst cases, if we copy one object in src bucket to dst bucket, the client has to GET to local disk and PUT to the dst bucket; If minIO supports sone COPY-like api, it can directly copy the object from src bucket to the dst bucket without transfering the object to the minIO client.
I wonder if minIO support this? Or how can I know where to learn about this? THANKS!

**Comment by klauspost** [comment]
MinIO supports the [CopyObject](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CopyObject.html) API.

**Reply by Wzing0421** [reply]
Hey, thank you so much for your reply!
Did you mean, By Copying Object I do not need to GET object to local disk and PUT to the dest bucket,  COPY Object means server-side copy, and the minIO server can COPY object from src bucket to dst bucket?
Thanks again for your answer!

**Reply by Wzing0421** [reply]
And another question, how to prove that?

**Reply by klauspost** [reply]
Read the API documentation, it is all in there.
